src/PCLExt.FileStorage.UWP/StorageFileImplementation.cs

```diff
@@ -0,0 +1,223 @@
+﻿using PCLExt.FileStorage.Extensions;
+using PCLExt.FileStorage.UWP.Extensions;
+using System;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+using Windows.Storage;
+
+namespace PCLExt.FileStorage.UWP
+{
+    /// <inheritdoc />
+    sealed class StorageFileImplementation : IFile
+    {
+        public string Name => _storageFile.Name;
+
+        public string Path => _storageFile.Path;
+
+        private bool _exists = false;
+        public bool Exists => _exists;
+
+        public long Size =>
+            (long)_storageFile.GetBasicPropertiesAsync().AsTask().Result.Size;
```

Comment:
Calling async code in properties is really not what we want.
http://blog.stephencleary.com/2013/01/async-oop-3-properties.html

Reply:
I'll fix it

Reply:
In according to the http://blog.stephencleary.com/2013/01/async-oop-3-properties.html the best way for Properties calling async methods is to return Task<T>. Do you mind if I change the interface of property? 
Example =>  public long Size -> public Task<long> Size
The implementation will be something like this:
public Task<long> Sizee => CalculateSize();

private async Task<long> CalculateSize()
{
   var props = await _storageFile.GetBasicPropertiesAsync();
   return (long)props.Size
}

The same for dates.

Reply:
No, this is not acceptable. My opinion is it's too counterintuitive

Reply:
I saw some examples of those task extensions like this
https://github.com/PokeD/PokeD.Server.NetCore/blob/5a0210e31bedfcf07630ecb4a4fa433b497dc21e/Extensions/TaskExtension.cs#L9
But with better handling of exceptions or something. Maybe this for now?

Reply:
As far as I see it's the same as I wrote initially:
(long)_storageFile.GetBasicPropertiesAsync().AsTask().Result.Size
the difference only that one allows to use CancellationToken but in this case it will be default :) and catching/re-throwing exceptions. And if I leave it as is the user will get exception anyway.

Reply:
Will it be an AggregateException?

Reply:
task.Wait()
task.Result
is the same as
task.Result
should we care about exceptions here or leave it to user if an exception will be thrown. Or may be it will be over coding.
And as for me it doesn't matter. Change to TResult Wait<TResult>(this Task<TResult> task, CancellationTokenSource cancellationTokenSource) or leave it as is?

Reply:
Yes it will be an AggregateException

Reply:
https://stackoverflow.com/a/17284612
Guess we should use ```GetAwaiter().GetResult()```

Reply:
We will block the execution this way but I'm not sure if we won't cause a deadlock this way, don't remember much of the async behaviour here